apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c

```diff
@@ -320,64 +822,181 @@ static void itc_audio_pcm_frames_to_bytes_tc_p(void)
 
 	bytes = pcm_frames_to_bytes(g_pcm, frame_size);
 	TC_ASSERT_GT_CLEANUP("pcm_frames_to_bytes", bytes, 0, pcm_close(g_pcm));
+
 	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
 	TC_SUCCESS_RESULT();
 }
 
 /**
-* @testcase         audio_pcm_bytes_to_frames_p
+* @testcase         itc_audio_pcm_frames_to_bytes_n_after_close
+* @brief            convert frames to bytes
+* @scenario         get configuration value of pcm and calculate frame to byte after pcm close
+* @apicovered       pcm_frames_to_bytes
+* @precondition     pcm should be opened and closed before
+* @postcondition    NA
+*/
+static void itc_audio_pcm_frames_to_bytes_n_after_close(void)
+{
+	ssize_t size = 0;
+	unsigned int bytes = 0;
+	unsigned int frame_size = 0;
+
+	g_pcm = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+
+	/* set basic configuration values for next test */
+	size = pcm_get_buffer_size(g_pcm);
+	TC_ASSERT_GT_CLEANUP("pcm_get_buffer_size", size, 0, pcm_close(g_pcm));
+
+	frame_size = pcm_bytes_to_frames(g_pcm, size);
+	TC_ASSERT_GT_CLEANUP("pcm_bytes_to_frames", frame_size, 0, pcm_close(g_pcm));
+
+	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
+
+	bytes = pcm_frames_to_bytes(g_pcm, frame_size);
+	TC_ASSERT_EQ("pcm_frames_to_bytes", bytes, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_frames_to_bytes_n
+* @brief            convert frames to bytes
+* @scenario         get configuration value of NULL pcm and calculate frame to byte
+* @apicovered       pcm_frames_to_bytes
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_audio_pcm_frames_to_bytes_n(void)
+{
+	unsigned int bytes = 1;
+	bytes = pcm_frames_to_bytes(NULL, pcm_get_buffer_size(g_pcm));
+	TC_ASSERT_EQ("pcm_frames_to_bytes", bytes, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_bytes_to_frames_p
 * @brief            convert bytes to frame
 * @scenario         get configuration value of pcm and calculate byte to frame
 * @apicovered       pcm_bytes_to_frames
 * @precondition     pcm should be opened before.
 * @postcondition    NA
 */
-static void itc_audio_pcm_bytes_to_frames_tc_p(void)
+static void itc_audio_pcm_bytes_to_frames_p(void)
 {
 	unsigned int frame_size = 0;
 	ssize_t size = 0;
 
+	g_pcm = NULL;
 	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
 	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
 
 	size = pcm_get_buffer_size(g_pcm);
```

Comment:
pcm_get_buffer_size returns the buffer size in terms of number of frames and not in number of bytes which is set by config parameters. So the tc can fail if the config parameters are not taken care of properly.